python/paddle/distribution/geometric.py

```diff
@@ -0,0 +1,340 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numbers
+
+import numpy as np
+
+import paddle
+from paddle.distribution import distribution, uniform
+from paddle.fluid import framework
+
+
+class Geometric(distribution.Distribution):
+    r"""
+    Geometric distribution parameterized by probs.
+
+    In probability theory and statistics, the geometric distribution is one of
+    discrete probability distributions, parameterized by one positive shape parameter, denoted by probs.
+    In n Bernoulli trials, it takes k trials to get the probability of success for the first time.
+    In detail, it is: the probability that the first k-1 times failed and the kth time succeeded.
+    The geometric distribution is a special case of the Pascal distribution when r=1.
+
+    The probability mass function (pmf) is
+
+    .. math::
+            Pr(Y=k)=(1-p)^kp
+
+    where k is number of trials performed and p is probability of success for each trial and k=0,1,2,3,4..., p belong to (0,1].
+
+    Args:
+        probs (Real|Tensor): Probability parameter.
+            The value of probs must be positive. When the parameter is a tensor, probs is probability of success for each trial.
+
+    Examples:
+
+        .. code-block:: python
+
+            import paddle
+            from paddle.distribution import Geometric
+
+            geom = Geometric(0.5)
+
+            geom.mean
+            # Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
+            #        [2.])
+
+            geom.variance
+            # Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
+            #        [2.])
+
+            geom.stddev
+            # Tensor(shape=[1], dtype=float32, place=Place(cpu), stop_gradient=True,
+            #        [1.41421354])
+    """
+
+    def __init__(self, probs=None):
```

Comment:
`probs` should not be a parameter with default value?

Reply:
> `probs` should not be a parameter with default value `None`?

Yes, here really should not set a default value for probs, i have fixed.

thanks for your review!